### PR TITLE
impl(bigquerycontrol): promote from experimental to transitional

### PR DESCRIPTION
**Pull request by scotthart**
This change is [Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14887)

### CODE REVIEW

**Review by dbolduc — COMMENTED** [review]
This is probably a convenient way to keep defining the `experimental-*` target, which `quickstart-cmake-pr` needs.

https://github.com/coryan/google-cloud-cpp/blob/49b7952a640dd4e323b2acafe18088557ff8f3a7/google/cloud/accessapproval/config.cmake.in#L23-L26

File: google/cloud/bigquerycontrol/README.md [hunk 1]
```diff
@@ -7,10 +7,8 @@ This directory contains an idiomatic C++ client library for the
 
 A data platform for customers to create, manage, share and query data.
 
-This library is **experimental**. Its APIs are subject to change without notice.
```
[the remark this answers is not in the document]
Comment (scotthart) [reply google/cloud/bigquerycontrol/README.md:10]: Done.

**Review by scotthart — COMMENTED** [review]
So the edits to the config.cmake.in fixed one issue, but because the quickstart-cmake build uses the current release and not HEAD to install cmake targets, the build could not find `experimental-bigquerycontrol`, so I have temporarily excluded it from that build.

> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14887#-:-ODlRVDf4O-8_uedG1qP:b-uttjy8)* status: 0 of 13 files reviewed, 1 unresolved discussion (waiting on @dbolduc)

File: google/cloud/bigquerycontrol/quickstart/BUILD.bazel [hunk 2]
```diff
@@ -20,6 +20,6 @@ cc_binary(
         "quickstart.cc",
     ],
     deps = [
-        "@google_cloud_cpp//:experimental-bigquerycontrol",
```
Comment (dbolduc) [inline google/cloud/bigquerycontrol/quickstart/BUILD.bazel:23]: I don't think we should be changing the quickstart targets just yet

https://github.com/googleapis/google-cloud-cpp/blob/main/doc/contributor/howto-guide-declare-a-library-ga.md#reference-the-ga-targets-in-the-quickstarts

File: google/cloud/bigquerycontrol/quickstart/CMakeLists.txt [hunk 3]
```diff
@@ -29,4 +29,4 @@ endif ()
 
 # Define your targets.
 add_executable(quickstart quickstart.cc)
-target_link_libraries(quickstart google-cloud-cpp::experimental-bigquerycontrol)
+target_link_libraries(quickstart google-cloud-cpp::bigquerycontrol)
```
Comment (dbolduc) [inline google/cloud/bigquerycontrol/quickstart/CMakeLists.txt:32]: same

File: google/cloud/bigquerycontrol/quickstart/BUILD.bazel (hunk 2 above)
Comment (scotthart) [reply google/cloud/bigquerycontrol/quickstart/BUILD.bazel:23]: Reverted.

File: google/cloud/bigquerycontrol/quickstart/CMakeLists.txt (hunk 3 above)
Comment (scotthart) [reply google/cloud/bigquerycontrol/quickstart/CMakeLists.txt]: Reverted.

**Review by scotthart — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/googleapis/google-cloud-cpp/14887)* status: 0 of 16 files reviewed, 3 unresolved discussions (waiting on @dbolduc)

File: google/cloud/spanner/samples/samples.cc [hunk 4]
```diff
@@ -1747,7 +1747,7 @@ void RestoreDatabaseWithEncryptionKeyCommand(std::vector<std::string> argv) {
 // [START spanner_restore_backup_with_MR_CMEK]
 void RestoreDatabaseWithMRCMEK(
     google::cloud::spanner_admin::DatabaseAdminClient client,
-    BackupIdentifier src, std::string const& database_id,
+    BackupIdentifier const& src, std::string const& database_id,
```
Comment (dbolduc) [inline google/cloud/spanner/samples/samples.cc:1750]: odd that we are making a spanner change in this PR, but it looks like a good change

**Review by dbolduc — APPROVED** [review]

**Pull request merged**